src/ocean/task/TaskPool.d

```diff
@@ -181,6 +185,9 @@ class TaskPool ( TaskT : Task ) : ObjectPool!(Task)
 
         foreach (task; tasks_iterator)
         {
+            enforce(!this.isSame(this.toItem(current_task), this.toItem(task)),
```

Comment:
Yeah, this PR introduces minor performance pessimization to verify the sanity condition. Should be fine for its use case but worth keeping in mind (maybe even put an explicit comment about it, @gkotian ?)